src/wasm.rs

```diff
@@ -44,16 +50,20 @@ impl HttpClient for WasmClient {
             }
 
             Ok(response)
-        });
-
-        Box::pin(InnerFuture { fut })
+        })
     }
 }
 
 struct InnerFuture {
     fut: Pin<Box<dyn Future<Output = Result<Response, Error>> + 'static>>,
 }
 
+impl InnerFuture {
+    fn new<F: Future<Output = Result<Response, Error>> + 'static>(fut: F) -> Pin<Box<Self>> {
+        Box::pin(Self { fut: Box::pin(fut) })
+    }
+}
+
```

Comment:
Instead of using `InnerFuture` we should probably use [send_wrapper](https://docs.rs/send_wrapper/0.4.0/send_wrapper/) instead. I think we can then convert the `send` method's body to be along the lines of:

```rust
async fn send(&self, req: Request) -> Result<Response, Error> {
    let fut = SendWrapper::new(async move {
        // perform the `fetch` call here
    });

    fut.await
}

Reply:
I haven't been able to figure out how to make this work with send_wrapper yet. Maybe that's a distinct PR?